pkg/kubectl/cmd/apply_view_last_applied.go

```diff
@@ -0,0 +1,163 @@
+		Long:    applyViewLastAppliedLong,
+		Example: applyViewLastAppliedExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, args))
+			cmdutil.CheckErr(options.Validate(cmd))
+			cmdutil.CheckErr(options.RunApplyViewLastApplied())
+		},
+	}
+
+	cmdutil.AddOutputVarFlagsForMutation(cmd, &options.OutputFormat)
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.")
+	usage := "that contains the last-applied-configuration annotations"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+
+	return cmd
+}
+
+func (o *ViewLastAppliedOptions) Complete(f cmdutil.Factory, args []string) error {
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(enforceNamespace, args...).
+		SelectorParam(o.Selector).
+		Latest().
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		configString, err := kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return err
+		}
+		if configString == nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("no last-applied-configuration annotation found on resource: %s\n", info.Name), info.Source, err)
+		}
+		o.LastAppliedConfigurationList = append(o.LastAppliedConfigurationList, string(configString))
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *ViewLastAppliedOptions) Validate(cmd *cobra.Command) error {
+	return o.validatePrinter(cmd)
+}
+
+func (o *ViewLastAppliedOptions) RunApplyViewLastApplied() error {
+	for _, str := range o.LastAppliedConfigurationList {
+		yamlOutput, err := yaml.JSONToYAML([]byte(str))
+		switch o.OutputFormat {
+		case "json":
+			jsonBuffer := &bytes.Buffer{}
+			err = json.Indent(jsonBuffer, []byte(str), "", "  ")
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(o.Out, string(jsonBuffer.Bytes()))
+		case "yaml":
+			fmt.Println(string(yamlOutput))
+			fmt.Fprintf(o.Out, string(yamlOutput))
+		}
+	}
+
+	return nil
+}
+
+func (o *ViewLastAppliedOptions) validatePrinter(cmd *cobra.Command) error {
+	format := cmdutil.GetFlagString(cmd, "output")
+	switch format {
+	case "json":
+		o.OutputFormat = "json"
+		return nil
+	// If flag -o is not specified, use yaml as default
+	case "yaml", "":
+		o.OutputFormat = "yaml"
+		return nil
+	default:
+		return cmdutil.UsageError(cmd, "The flag 'output' must be one of yaml|json")
```

Comment:
What does the help command show when you run kubectl apply view-last-applied -h?  Does it list the other format types?